### PR TITLE
Restrict aichat controller to pilot

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR restricts the access to `chat_completion` of `aichat_controller` to users in the `SingleUserExperiment` with pilot name`gen-ai-customizing-llm` and students of teachers in the pilot.

Note that I added `has_required_params` to `aichiat_controller` so that a `bad_request` response would be returned to the client instead of just erroring out. I also included dashboard unit tests.

## Warning!!

The [AP CSP Create Performance Task](https://apcentral.collegeboard.org/courses/ap-computer-science-principles/exam) is in progress. The most critical dates are from April 3 - April 30, 2024. Please consider any risk introduced by this PR that could affect our students taking AP CSP. Code.org students taking AP CSP primarily use App Lab for their Create Task, however a small percent use Game Lab. Carefully consider whether your change has any risk of alterering, changing, or breaking anything in these two labs. Even small changes, such as a different button color, are considered significant during this time period. Reach out to the Student Learning team or Curriculum team for more details.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/ability.rb [hunk 1]
```diff
@@ -2,8 +2,7 @@ class Ability
   include CanCan::Ability
   include Pd::Application::ActiveApplicationModels
 
-  CSA_PILOT = 'csa-pilot'
-  CSA_PILOT_FACILITATORS = 'csa-pilot-facilitators'
+  GENAI_PILOT = 'genai-pilot'
```
Comment (fisher-alice) [inline dashboard/app/models/ability.rb]: Confirming name with @dancodedotorg [Slack thread](https://codedotorg.slack.com/archives/C06FELVTV0Q/p1713210562314509).

File: dashboard/app/models/ability.rb [hunk 2]
```diff
@@ -2,8 +2,7 @@ class Ability
   include CanCan::Ability
   include Pd::Application::ActiveApplicationModels
 
-  CSA_PILOT = 'csa-pilot'
-  CSA_PILOT_FACILITATORS = 'csa-pilot-facilitators'
```
Comment (fisher-alice) [inline dashboard/app/models/ability.rb:6]: I believe these are no longer used.

**Review by thomasoniii — APPROVED** [review]
lgtm 🚀

File: dashboard/test/controllers/aichat_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    valid_message = "hello"
+    pii_violation_message = "my email is l.lovepadel@sports.edu"
+    profanity_violation_message = "Damn you, robot"
+    @valid_params = @params.merge(newMessage: valid_message)
+    @pii_violation_params = @params.merge(newMessage: pii_violation_message)
+    @profanity_violation_params = @params.merge(newMessage: profanity_violation_message)
+  end
+
+  test_user_gets_response_for :chat_completion,
```
Comment (bencodeorg) [inline dashboard/test/controllers/aichat_controller_test.rb:25]: Nice tests!

**Review by bencodeorg — APPROVED** [review]
🎉

**Pull request merged**